scripts/subprocess_utils.py

```diff
@@ -0,0 +1,157 @@
+import os
+import sqlite3
+from typing import Optional
+
+import subprocess
+import psutil
+
+STDOUT_LOG = "/commit_reveal_stdout.log"
+STDERR_LOG = "/commit_reveal_stderr.log"
+PROCESS_NAME = "commit_reveal.py"
+
+
+def is_process_running(process_name: str) -> bool:
+    """Check if a process with a given name is currently running."""
+    for proc in psutil.process_iter(['pid', 'name', 'cmdline']):
+        cmdline = proc.info['cmdline']
+        if cmdline and (process_name in proc.info['name'] or any(process_name in cmd for cmd in cmdline)):
+            return True
+    return False
+
+
+def get_process(process_name: str) -> Optional[int]:
+    """Check if a process with a given name is currently running, and return its PID if found."""
+    for proc in psutil.process_iter(['pid', 'name', 'cmdline']):
+        cmdline = proc.info['cmdline']
+        if cmdline and (process_name in proc.info['name'] or any(process_name in cmd for cmd in cmdline)):
+            return proc.info['pid']
+    return None
+
+
+def start_commit_reveal_subprocess(network: Optional[str] = None, sleep_interval: Optional[float] = None):
+    """Start the commit reveal subprocess if not already running."""
+    log_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "subprocess", "logs"))
```

Comment:
I understand, but I think it might make more sense to store the logs where the process lives. I am kind of split on this one.

Reply:
Honestly, I don't really understand how this can be useful, since most users won't even know where the process is located. But the entire community knows where we store all the bittensor, wallet, and cli related files. This is a historically established location. I suggest being consistent (as you suggest me many times ;) ).

Reply:
If you are the type to be concerned about a process then you know how to look them up and see what they do. In our case its extremely clear where the process originated and even what parameters where passed to it

![image](https://github.com/user-attachments/assets/c5ffa85c-e929-4de4-a470-9734281cadfc)

Reply:
I see. But let's use `~/.bittensor/` directory for storing all data `bittensor` SDK related.

Reply:
Done. Storing in ~/.bittensor/logs directory